beacon_node/beacon_chain/src/builder.rs

```diff
@@ -912,6 +924,8 @@ where
             validator_pubkey_cache: TimeoutRwLock::new(validator_pubkey_cache),
             attester_cache: <_>::default(),
             early_attester_cache: <_>::default(),
+            lightclient_server_cache: LightclientServerCache::new(),
```

Comment:
Can we make this an `Option`  so we only compute and cache the proofs if light client server is enabled?

Reply:
I agree that conceptually looks cleaner however adding an option requires to handle the case where HTTP API and ReqResp handlers receive a request but the lightclient server in disabled. We can achieve the same purpose by only adding data into the lightclient_server_cache if a config value is true. In the case the lightclient server is disabled there will be an extra cache in memory, but I think overall it reduces complexity. Let's just make sure that the network flag disabling the lightclient routes is the also used in the BeaconChain config.

Reply:
Yeah cool, I'm happy either way as long as we don't perform the computation if light client server is disabled.